python/lsst/pipe/base/pipeline_graph/_pipeline_graph.py

```diff
@@ -1093,32 +842,300 @@ def remove_task_subset(self, subset_label: str) -> None:
+        attributes added.
+        """
+        bipartite_xgraph = self._make_bipartite_xgraph_internal(init)
+        task_keys = [
+            key
+            for key, bipartite in bipartite_xgraph.nodes(data="bipartite")
+            if bipartite == NodeType.TASK.bipartite
+        ]
+        return self._transform_xgraph_state(
+            networkx.algorithms.bipartite.projected_graph(networkx.DiGraph(bipartite_xgraph), task_keys),
+            skip_edges=True,
+        )
+
+    def make_dataset_type_xgraph(self, init: bool = False) -> networkx.DiGraph:
+        """Return a networkx representation of just the dataset types in the
+        pipeline.
+
+        Parameters
+        ----------
+        init : `bool`, optional
+            If `True` (`False` is default) return the graph of init input and
+            output dataset types, instead of the graph of runtime (input,
+            output, prerequisite input) dataset types.
+
+        Returns
+        -------
+        xgraph : `networkx.DiGraph`
+            Directed acyclic graph with no parallel edges.
+
+        Notes
+        -----
+        The returned graph uses `NodeKey` instances for nodes.  The tasks that
+        link these tasks are not represented at all; edges have no attributes,
+        and there are no parallel edges.
+
+        See `DatasetTypeNode` for the descriptive node and attributes added.
+        """
+        bipartite_xgraph = self._make_bipartite_xgraph_internal(init)
+        dataset_type_keys = [
+            key
+            for key, bipartite in bipartite_xgraph.nodes(data="bipartite")
+            if bipartite == NodeType.DATASET_TYPE.bipartite
+        ]
+        return self._transform_xgraph_state(
+            networkx.algorithms.bipartite.projected_graph(
+                networkx.DiGraph(bipartite_xgraph), dataset_type_keys
+            ),
+            skip_edges=True,
+        )
+
+    ###########################################################################
+    #
+    # Serialization Interface.
+    #
+    ###########################################################################
+
+    @classmethod
+    def read_stream(
+        cls,
+        stream: BinaryIO,
+        import_and_configure: bool = True,
+        check_edges_unchanged: bool = False,
+        assume_edges_unchanged: bool = False,
+    ) -> PipelineGraph:
+        """Read a serialized `PipelineGraph` from a file-like object.
+
+        Parameters
+        ----------
+        stream : `BinaryIO`
+            File-like object opened for binary reading, containing
+            gzip-compressed JSON.
+        import_and_configure : `bool`, optional
+            If `True`, import and configure all tasks immediately (see the
+            `import_and_configure` method).  If `False`, some `TaskNode` and
+            `TaskInitNode` attributes will not be available, but reading may be
+            much faster.
+        check_edges_unchanged : `bool`, optional
```

Comment:
I've now pushed a version with this change.  It does not use the new enum in `reconfigure_tasks`, since there are fewer options in play there and the current state seems less confusing.